Adaptors/Amqp/src/PullQueueStorage.cs

```diff
@@ -116,10 +113,15 @@ public override async Task Init(CancellationToken cancellationToken)
     using var _               = logger_!.LogFunction();
     var       nbPulledMessage = 0;
 
+    if (!IsInitialized)
+    {
+      throw new ArmoniKException($"{nameof(PullQueueStorage)} should be initialized before calling this method.");
+    }
+
     while (nbPulledMessage < nbMessages)
     {
       var currentNbMessages = nbPulledMessage;
-      for (var i = receivers_.Length - 1; i >= 0; --i)
+      for (var i = receivers_!.Length - 1; i >= 0; --i)
       {
         cancellationToken.ThrowIfCancellationRequested();
         var receiver = await receivers_[i];
```

Comment:
As `Init` is always called at this point, maybe `receivers_` and `senders_` could be directly a list `ReceiverLink` and `SenderLink` instead of tasks. This would avoid the `await` here.